corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -290,7 +292,19 @@ hqDefine("cloudcare/js/form_entry/form_ui", [
                     return options.target;
                 },
                 key: function (data) {
-                    return ko.utils.unwrapObservable(data.uuid) || ko.utils.unwrapObservable(data.ix);
+                    let key = ko.utils.unwrapObservable(data.uuid);
+                    if (key) {
+                        return key;
+                    }
+                    key = ko.utils.unwrapObservable(data.ix);
+
+                    const isRepeatDummy = data.type === constants.GROUP_TYPE && data.exists === "false";
+                    if (isRepeatDummy) {
+                        // Without this a key that was previously used by a group would be used by the dummy.
+                        // This causes issues because the template and other fields would not be reset.
+                        key = key.substring(0, key.lastIndexOf('_') + 1) + 'r';
```

Comment:
Just curious, where does this key format, with the ending `r`, come from? Is this a convention formplayer uses?

Reply:
Formplayer uses numbers I just wanted something unique. But I now realize this is not necessary anymore with the other fix for deleting groups in.